diff --git a/deploy/actions.js b/deploy/actions.js
index 8aeff80706c60ce1108621f7c971c8dfa83e3120..e4c3718afe7b5b3619659bbff912dc677f43f9fa 100644
--- a/deploy/actions.js
+++ b/deploy/actions.js
@@ -29,10 +29,10 @@ function deploy(engine, context, buildTarget, options) {
             }
         }
         const buildOptions = yield context.getTargetOptions(architect_1.targetFromTargetString(buildTarget.name));
-        if (!buildOptions.outputPath || typeof buildOptions.outputPath !== 'string') {
+        if (!options.outputPath && !buildOptions.outputPath) {
             throw new Error(`Cannot read the output path option of the Angular project '${buildTarget.name}' in angular.json`);
         }
-        yield engine.run(buildOptions.outputPath, options, context.logger);
+        yield engine.run(options.outputPath ?? buildOptions.outputPath, options, context.logger);
     });
 }
 exports.default = deploy;
diff --git a/deploy/schema.json b/deploy/schema.json
index 33fd668a1c57e62a893ca9a142599275f50c32b9..6c79b72fbdde752170048ea9cbda0685a5216dc2 100644
--- a/deploy/schema.json
+++ b/deploy/schema.json
@@ -48,6 +48,10 @@
       "description": "Includes dotfiles by default. Execute with --no-dotfiles to ignore files starting with `.`.",
       "default": false
     },
+    "outputPath": {
+      "type": "string",
+      "description": "The location of the build output."
+    },
     "cname": {
       "type": "string",
       "description": "Generate a CNAME file for the specified domain.",